### DISCUSSION TITLE
Cache on file change

### DISCUSSION THREAD

**Discussion by dgrempka** [imgproxy/imgproxy#1101] [Q&A] [answered]
Hi,
I'm playing around with imageproxy with an local image storage and run into the problem that if I update my image, the cache would still show me the previous image.
I couldn't find any solutions in the documentation about cache, except lowering the expired timestamp.
Currently I have the options by renaming the file to generate a new url or by deleting the whole cache.

Is there an option to delete all connected cached files of a local image without deleting the whole cache and maybe even on file updates?

**Comment by DarthSim** [comment, marked as answer]
Since imgproxy doesn't cache results and relies on a third-party cache, it can't control when the cache will request a new version of an image. However, imgproxy has a `cachebuster` option that allows changing an URL without affecting the processing flow.

The common practice is to not store changed files under the same name. Using the same name for changed files may affect the cache on different levels. For example, the browser will not request a new version of a file until its cache is expired. That's one of the reasons why web app builders usually add hash to `.css` and `.js` assets.

**Comment by dgrempka** [comment]
Thanks for the clarification!
